File: yoojongho/2024-08-04-complement-of-base-10-integer.java
/**
 * Leetcode
 * problem: 1009
 * link: https://leetcode.com/problems/complement-of-base-10-integer/description/
 * tag: Bit Manipulation
 */

class Solution {
    public int bitwiseComplement(int n) {
        if(n == 0) return 1;
        int result = 0;
        int counter = 0;
        while (n > 0){
            int digit = n % 2;
            if(digit == 1) digit = 0;
            else digit = 1;
            result += digit * Math.pow(2, counter);
            n /= 2;
            counter++;
        }
        return result;
    }
}

File: Changwhi/2024-08-04- Complement-of-Base-10-Integer.py
class Solution:
    def bitwiseComplement(self, n: int) -> int:
        cnt=0
        ans=0
        if n==0:
            return 1
        while n > 0:
            if n&1:
                cnt+=1
            else:
                ans =ans +(2**cnt)
                cnt+=1
            n=n>>1
        return ans